src/backend/utils/cache/plancache.c

```diff
@@ -856,6 +860,41 @@ CheckCachedPlan(CachedPlanSource *plansource)
 	return false;
 }
 
+/*
+ * detect_plangen_switch: check if plangen switched between statements.
+ *
+ * Check if plangen switched for statements of one plan or between plans.
+ * 'plan' should point to currently initialized plan.
+ * 'plansource' is where plangen summary stored for all plans.
+ */
+static void
+detect_plangen_switch(CachedPlan *plan, CachedPlanSource *plansource)
+{
+	if (plansource->plangen_switched || list_length(plan->stmt_list) == 0)
+		return;
+
+	ListCell	*lc = list_head(plan->stmt_list);
+	PlannedStmt	*plannedstmt;
```

Comment:
It seems that C89 rule of declaring variables at the beginning of a scope block is not followed anymore.